easybuild/easyconfigs/o/OpenCV/OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb

```diff
@@ -0,0 +1,46 @@
+name = 'OpenCV'
+version = '3.4.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://opencv.org/'
+description = """OpenCV (Open Source Computer Vision Library) is an open source computer vision
+ and machine learning software library. OpenCV was built to provide
+ a common infrastructure for computer vision applications and to accelerate
+ the use of machine perception in the commercial products."""
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+
+source_urls = [
+    'https://github.com/Itseez/opencv/archive/',
+    # the hash is version dependent! see 3rdparty/ippicv/ippicv.cmake
+    'https://raw.githubusercontent.com/Itseez/opencv_3rdparty/dfe3162c237af211e98b8960018b564bc209261d/ippicv',
+]
+sources = [
+    '%(version)s.zip',
+    {'filename': 'ippicv_2017u3_lnx_intel64_general_20170822.tgz', 'extract_cmd': "cp %s %(builddir)s"},
+]
+checksums = [
+    '24bda432eaace9e992322dcc3d30144cefa5314c2424d4aa02e5fe3fa9dd17bd',  # 3.4.1.zip
+    # ippicv_2017u3_lnx_intel64_general_20170822.tgz
+    '9a4b14a24d31768c3ead0720f27d55dcf80723b5ba1cd8903bed1fd69f1b9e2e',
+]
+
+builddependencies = [
+    ('CMake', '3.10.2'),
+]
+
+dependencies = [
+    ('Python', '2.7.14'),
+    ('zlib', '1.2.11'),
+    ('FFmpeg', '3.4.2'),
+    ('libjpeg-turbo', '1.5.3'),
+    ('libpng', '1.6.34'),
+    ('LibTIFF', '4.0.9'),
+    ('JasPer', '2.0.14'),
+    ('Java', '1.8.0_162', '', True),
```

Comment:
Please use the `('Java', '1.8')` alias if possible

Reply:
@vanzod I didn't manage to make the `('Java', '1.8')` alias work.  The error I get is `OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb: Failed to determine minimal toolchain for dep Java 1.8`.  I'm happy to have another go if you can suggest how to proceed.

Reply:
@SimonPinches My bad, you still need to tell EB to consider the software built with `dummy` toolchain in order to pick up the `Java` wrapper. Try with `('Java', '1.8', '', True)`

Reply:
@vanzod Still no joy :-(   Seems like the `_162` is not correctly picked up for `Java` or `ant`.
```
$ eb OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb -f --robot-paths=/home/ITER/pinches/local/easybuild-easyconfigs-clean/easybuild/easyconfigs:
== temporary log file in case of crash /tmp/eb-3KUe5v/easybuild-GwfE51.log
== processing EasyBuild easyconfig /home/ITER/pinches/local/easybuild-easyconfigs-clean/easybuild/easyconfigs/o/OpenCV/OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb
== building and installing OpenCV/3.4.1-intel-2018a-Python-2.7.14...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== FAILED: Installation ended unsuccessfully (build directory: /work/imas/opt/EasyBuild/build/OpenCV/3.4.1/intel-2018a-Python-2.7.14): build failed (first 300 chars): Missing modules for dependencies (use --robot?): Java/1.8.0, ant/1.10.1-Java-1.8.0
== Results of the build can be found in the log file(s) /tmp/eb-3KUe5v/easybuild-OpenCV-3.4.1-20190503.152704.YxCyO.log
ERROR: Build of /home/ITER/pinches/local/easybuild-easyconfigs-clean/easybuild/easyconfigs/o/OpenCV/OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb failed (err: 'build failed (first 300 chars): Missing modules for dependencies (use --robot?): Java/1.8.0, ant/1.10.1-Java-1.8.0')
```

Reply:
@SimonPinches It looks you specified the Java version as `1.8.0` instead of `1.8`. Can you confirm?

Reply:
@vanzod Well spotted :-)  The next problem is getting the correct `Java` dependency for `ant`.  Am I specifying this correctly in the `OpenCV` `eb` file as `('ant', '1.10.1', '-Java-%(javaver)s', True),`?
```
$ eb ../../o/OpenCV/OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb --robot-paths=/home/ITER/pinches/local/easybuild-easyconfigs-clean/easybuild/easyconfigs:
== temporary log file in case of crash /tmp/eb-5tl9Ka/easybuild-D81wlh.log
== processing EasyBuild easyconfig /home/ITER/pinches/local/easybuild-easyconfigs-clean/easybuild/easyconfigs/o/OpenCV/OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb
== building and installing OpenCV/3.4.1-intel-2018a-Python-2.7.14...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== FAILED: Installation ended unsuccessfully (build directory: /work/imas/opt/EasyBuild/build/OpenCV/3.4.1/intel-2018a-Python-2.7.14): build failed (first 300 chars): Missing modules for dependencies (use --robot?): Java/1.8, ant/1.10.1-Java-1.8
== Results of the build can be found in the log file(s) /tmp/eb-5tl9Ka/easybuild-OpenCV-3.4.1-20190504.140111.zHBgW.log
ERROR: Build of /home/ITER/pinches/local/easybuild-easyconfigs-clean/easybuild/easyconfigs/o/OpenCV/OpenCV-3.4.1-intel-2018a-Python-2.7.14.eb failed (err: 'build failed (first 300 chars): Missing modules for dependencies (use --robot?): Java/1.8, ant/1.10.1-Java-1.8')
```

Reply:
@SimonPinches That's indeed the correct way of specifying `ant` (we already have exactly that in the `OpenCV` easyconfigs you're using as a base. The error you ran into is simply because you were not using `--robot`.

You'll need to stick to the `('Java', '1.8.0_162', '', True),` btw, because otherwise the tests may fail because you're using a different `Java` version as a dependency in the `2018a` generation of easyconfigs.

We've started consistently using the `Java/1.8` wrapper instead in the `2018b` generation of easyconfigs (see also https://easybuild.readthedocs.io/en/latest/Wrapping_dependencies.html), I guess @vanzod overlooked that.

With that in mind, this PR is good to go w.r.t. updates of already supported software (`OpenCV`, `PyQt5` and `SIP`).